web/react/components/team_general_tab.jsx

```diff
@@ -93,10 +244,178 @@ export default class GeneralTab extends React.Component {
             serverError = this.state.serverError;
         }
 
+        let teamListingSection;
+        if (this.props.activeSection === 'team_listing') {
+            const inputs = [
+                <div key='userTeamListingOptions'>
+                    <div className='radio'>
+                        <label>
+                            <input
+                                name='userTeamListingOptions'
+                                type='radio'
+                                defaultChecked={this.state.allow_team_listing}
+                                onChange={this.handleTeamListingRadio.bind(this, true)}
+                            />
+                            {'Yes'}
+                        </label>
+                        <br/>
+                    </div>
+                    <div className='radio'>
+                        <label>
+                            <input
+                                name='userTeamListingOptions'
+                                type='radio'
+                                defaultChecked={!this.state.allow_team_listing}
+                                onChange={this.handleTeamListingRadio.bind(this, false)}
+                            />
+                            {'No'}
+                        </label>
+                        <br/>
+                    </div>
+                    <div><br/>{'When allowed then the team will appear on the main page as part of team directory if team browsing is enabled in the system console.'}</div>
+                </div>
+            ];
+
+            teamListingSection = (
```

Comment:
Since this is disabled by the admin console by default, this should be "grayed out" when it is disabled there. Otherwise there will be confusion (even with the help text).